src/com.target.lorri.varlink

```diff
@@ -1,85 +1,58 @@
 # The interface `lorri daemon` exposes.
 interface com.target.lorri
 
-# WatchShell instructs the daemon to evaluate a Nix expression and re-evaluate
-# it when it or its dependencies change.
-method WatchShell(shell_nix: ShellNix) -> ()
-
-# ShellNix describes the Nix expression which evaluates to a development
-# environment.
-type ShellNix (
-  # The absolute path of a Nix file specifying the project environment.
-  path: string
-)
-
 # Monitor the daemon. The method will reply with an update whenever a build begins or ends.
 # Montior will immediately reply with a snapshot of known projects, then a marker event,
 # indicating that the stream of events is now "live."
 method Monitor() -> (event: Event)
 
 type Event (
     kind: (section_end, started, completed, failure),
-    nix_file: ?ShellNix, # only present if kind != section_end
-    reason: ?Reason,     # only present if kind == started
-    result: ?Outcome,    # only present if kind == completed
-    failure: ?Failure    # only present if kind == failure
+    section: ?SectionMarker, # iff kind == section_end
+    reason: ?Reason,        # iff kind == started
+    result: ?Outcome,       # iff kind == completed
+    failure: ?Failure       # iff kind == failure
 )
 
+
+type SectionMarker ()
+
 type Reason (
     kind: (project_added, ping_received, files_changed, unknown),
-    project: ?ShellNix, # only present if kind == project_added
-    files: ?[]string,   # only present if kind == files_changed
-    debug: ?string      # only present if kind == unknown
+    project: ?string, # iff kind == project_added
+    files: ?[]string, # iff kind == files_changed
+    debug: ?string    # iff kind == unknown
 )
 
 type Outcome (
+    nix_file: ?string,
     project_root: string
 )
 
 type Failure (
     kind: (io, spawn, exit, output),
-    msg: ?string,   # only present if kind in (io, spawn)
-    cmd: ?string,   # only present if kind in (spawn, exit)
-    status: ?int,   # only present if kind == exit
-    logs: ?[]string # only present if kind == exit
+    nix_file: string,
+    io: ?IOFail,        # iff kind == io
+    spawn: ?SpawnFail,   # iff kind == spawn
+    exit: ?ExitFail,     # iff kind == exit
+    output: ?OutputFail # iff kind == output
 )
 
-# WatchServices establishes a stream with the daemon. Initially, the daemon
-# evaluates the given services definition to an array of Command objects and
-# sends a reply for each of them. After this initial evaluation, the daemon
-# watches the services definition and its dependencies for changes,
-# re-evaluates it as appropriate and sends a reply for each Command again.
-#
-# This is a streaming RPC. The daemon only accepts client calls with the "more"
-# property set - see https://varlink.org/Method-Call.
-# TODO: Implement WatchServices
-#method WatchServices(services_nix: ServicesNix) -> (service: Service)
-
-# ServicesNix describes the Nix expression which evaluates to a list of
-# services.
-type ServicesNix (
```

Comment:
I used to design gRPC APIs a bunch. In doing so, I found the [Google API Design Guidelines](https://cloud.google.com/apis/design) a real help.

The style I settled on here is a combination of Google API Design Guidelines and the [Golang comment format](https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences).

Concretely, this means:
- every type has a comment
- every not completely trivial field has a comment
- the comment format for type or field `X` is `X <verb> <rest of at least one full sentence>.`
- type and field names are full English words unless they are very long and the abbreviation is very common, e.g. `api` instead of `application_programming_interface`

Particularly if this is to be the project's public API, I think it is worth making the declarations in this file look consistent and documenting it.

Reply:
The only quibble I'd make here is that the Go comment format (i.e. `X <verb> <sentence>`) arises in part because of the anemic feature set of godoc, and often leads to awkward and boring documentation. (Often I see `Read implements io.Reader on MyType`)

I'm currently adding documentation comments based on the rest of that guide, using the dictionary style that the Google guidelines examples use.

Reply:
> I'm currently adding documentation comments based on the rest of that guide, using the dictionary style that the Google guidelines examples use.

Awesome, thank you!